metals/src/main/scala/scala/meta/internal/parsing/TokenEditDistance.scala

```diff
@@ -312,11 +312,15 @@ object TokenEditDistance {
     } else if (originalInput.value.isEmpty() || revisedInput.value.isEmpty()) {
       noMatch
     } else {
+
+      // Tokenizing should work without issues in all Scala sources
+      val defaultDialect = scala.meta.dialects.Scala213
```

Comment:
Just an observation, we seem to define `defaultDialect` quite a few places around the codebase. I wonder if there is a way we can simplify this and set the default in one place?

Reply:
I thinking of doing `defaultTokenizingDialect` for now, since tokenizing code works ok for all languages - it just might not give you real tokens like `KwGiven`, but honestly it's not needed in most places.

Reply:
I changed msot places to use Trees.defaultTokenizerDialect. Any leftovers will be migrated once I manage to fix Mtags to use a correct versions, but I did not want to do it all in this one PR.